lib/components/Web3Context/index.tsx

```diff
@@ -1,19 +1,27 @@
-import React, { createContext, FC } from "react";
+import React, { createContext, FC, useState } from "react";
 
-import { useWeb3Modal } from "../../utils/useWeb3Modal";
-import { web3InitialState, Web3ModalState } from "./types";
+import { useProvider } from "../../utils/useProvider";
+import { ConnectModal } from "../ConnectModal";
+import { RenderModalProps, web3InitialState, Web3ModalState } from "./types";
 
 export const Web3Context = createContext<Web3ModalState>(web3InitialState);
 
 interface Props {
   children: React.ReactNode;
 }
+
 /** Init the web3Modal and expose via react context  */
 export const Web3ContextProvider: FC<Props> = (props) => {
-  const web3ProviderState = useWeb3Modal();
-
+  const providerState = useProvider();
+  const [showModal, setShowModal] = useState(false);
+  const toggleModal = () => setShowModal((s) => !s);
+  const renderModal = (props: RenderModalProps) => (
+    <ConnectModal {...props} showModal={showModal} />
+  );
```

Comment:
ah i just realized we should use useCallback here, so that this component only reinstantiates itself when `showModal` changes. Otherwise changes to `providerState` may cause it to fully unmount and remount which can cause scary hard-to-debug issues. ill drop that in and merge